wagtail/admin/templatetags/wagtailadmin_tags.py

```diff
@@ -1194,11 +1198,118 @@ class PanelNode(BlockInclusionNode):
+register.tag("rawformattedfield", RawFormattedFieldNode.handle)
+
+
+@register.inclusion_tag("wagtailadmin/shared/formatted_field.html")
+def formattedfield(
+    field=None,
+    rendered_field=None,
+    classname="",
+    show_label=True,
+    id_for_label=None,
+    sr_only_label=False,
+    icon=None,
+    help_text=None,
+    help_text_id=None,
+    show_add_comment_button=False,
+    label_text=None,
+    error_message_id=None,
+):
+    """
+    Renders a form field in standard Wagtail admin layout.
+    - `field` - The Django form field to render.
+    - `rendered_field` - The rendered HTML of the field, to be used in preference to `field`.
+    - `classname` - For legacy patterns requiring field-specific classes. Avoid if possible.
+    - `show_label` - Hide the label if it is rendered outside of the field.
+    - `id_for_label` - Manually set this this if the field’s HTML isn’t rendered by Django (for example hard-coded in HTML).
+        We add an id to the label so we can use it as a descriptor for the "Add comment" button.
+    - `sr_only_label` - Make the label invisible for all but screen reader users. Use this if the field is displayed without a label.
+    - `icon` - Some fields have an icon, though this is generally a legacy pattern.
+    - `help_text` - Manually set this if the field’s HTML is hard-coded.
+    - `help_text_id` - The help text’s id, necessary so it can be attached to the field with `aria-describedby`.
+    - `show_add_comment_button` - Display a comment control within Wagtail forms.
+    - `label_text` - Manually set this if the field’s HTML is hard-coded.
+    - `error_message_id` - ID of the error message container element.
+    """
+
+    label_for = id_for_label or (field and field.id_for_label) or ""
+
+    context = {
+        "classname": classname,
+        "show_label": show_label,
+        "sr_only_label": sr_only_label,
+        "icon": icon,
+        "show_add_comment_button": show_add_comment_button,
+        "error_message_id": error_message_id,
+        "label_for": label_for,
+        "label_id": f"{label_for}-label" if label_for else "",
+        "label_text": label_text or (field and field.label) or "",
+        "required": field and field.field.required,
+        "contentpath": field.name if field else "",
+        "help_text": help_text or (field and field.help_text) or "",
+    }
+
+    if help_text_id:
+        context["help_text_id"] = help_text_id
+    elif field and field.help_text and field.id_for_label:
+        context["help_text_id"] = f"{field.id_for_label}-helptext"
+    else:
+        context["help_text_id"] = ""
+
+    if field:
+        context["rendered_field"] = rendered_field or render_with_errors(field)
+        context[
+            "field_classname"
+        ] = f"w-field--{ fieldtype(field) } w-field--{ widgettype(field) }"
+
+        errors = field.errors
+        has_errors = bool(errors)
+        if has_errors and hasattr(field.field.widget, "render_with_errors"):
```

Comment:
Perhaps we should reuse the code for `has_unrendered_errors` here, to align with what the template originally had (and avoid the weirdness of `field.field` because `field` is technically a `BoundField`).

```suggestion
        if has_errors and not has_unrendered_errors(field):
```

Reply:
To be honest, the existence of `has_unrendered_errors` has always bugged me - a bit of one-off rendering logic to say "if this field has errors and doesn't handle rendering them for itself, render them here" that I would rather have placed directly in the template but couldn't, so had to come up with a tortuously-named filter to capture the idea. Being able to write that logic inline in actual Python code is scratching a very long-standing itch :-)

I can't remember why I stopped short of removing `has_unrendered_errors` completely here - maybe it was out of concern that third-party code might be using it. Either way, I'd be in favour of us deprecating it at the same time as deprecating direct use of field.html.

Reply:
OK, if the plan is to remove `has_unrendered_errors` then I'm all for it, as that's what I'd prefer too.